srcjs/input_binding_sidebarmenuExpanded.js

```diff
@@ -0,0 +1,56 @@
+/* global Shiny */
+
+// sidebarmenuExpandedInputBinding
+// ------------------------------------------------------------------
+// This keeps tracks of what menuItem (if any) is expanded
+var sidebarmenuExpandedInputBinding = new Shiny.InputBinding();
+$.extend(sidebarmenuExpandedInputBinding, {
+  find: function(scope) {
+    return $(scope).find('section.sidebar');
+  },
+  getId: function(el) {
+    return "itemExpanded";
+  },
+  // needed so we set the appropriate value for bookmarked apps on startup
+  initialize: function(el) {
+    $(this).trigger('change');
+  },
+  // the value is the href of the open menuItem (or NULL if there's
+  // no open menuItem)
+  getValue: function(el) {
+    var $expanded = $(el).find('li ul.menu-open');
+    if ($expanded.length === 1) {
+      return $expanded.find('a').attr('href').substring(1);
+    } else if ($(el).attr("data-expanded")) {
+      return $(el).attr("data-expanded");
+    } else {
+      return null;
+    }
+  },
+  setValue: function(el, value) {
+    if (value !== null) {
+      var firstChild = 'a[href="#' + value + '"]';
+      var $ul = $(firstChild).parent().parent('.treeview-menu');
+      $ul.addClass('menu-open');
```

Comment:
It might be simpler in the long run to check for the `menu-open` class, and if it's not present, trigger a click event on the `$(firstChild).parent().prev()` (or whatever it is), which will cause the event handler from AdminLTE to run. Then you don't have to try to duplicate the behavior, like calling `$ul.show()`.